tests/upgrades/test_subscription.py

```diff
@@ -30,161 +30,165 @@
+        upload_manifest(settings.fake_manifest.url['default'], org.name)
+        history = entities.Subscription(organization=org).manifest_history(
+            data={'organization_id': org.id}
+        )
+        assert f"{org.name} file imported successfully." == history[0]['statusMessage']
+        sub = entities.Subscription(organization=org)
+        sub.refresh_manifest(data={'organization_id': org.id})
+        assert len(sub.search()) > 0
+
+    @post_upgrade(depend_on=test_pre_manifest_scenario_refresh)
+    def test_post_manifest_scenario_refresh(self, request, dependent_scenario_name):
+        """After upgrade, Check the manifest refresh and delete functionality.
+
+        :id: postupgrade-29b246aa-2c7f-49f4-870a-7a0075e184b1
+
+        :steps:
+            1. Refresh manifest.
+            2. Delete manifest.
+
+        :expectedresults: After upgrade,
+            1. Pre-upgrade manifest should be refreshed and deleted.
+        """
+        pre_test_name = dependent_scenario_name
+        org = entities.Organization().search(query={'search': f'name={pre_test_name}_org'})[0]
+        request.addfinalizer(org.delete)
+        sub = entities.Subscription(organization=org)
+        sub.refresh_manifest(data={'organization_id': org.id})
+        assert len(sub.search()) > 0
+        delete_manifest(org.name)
+        history = entities.Subscription(organization=org).manifest_history(
+            data={'organization_id': org.id}
+        )
+        assert "Subscriptions deleted by foreman_admin" == history[0]['statusMessage']
+
+
+class TestSubscriptionAutoAttach:
     """
-    pre_test_name = [
-        mark.kwargs['depend_on'].__name__
-        for mark in request.node.own_markers
-        if 'depend_on' in mark.kwargs
-    ][0]
-    org = entities.Organization().search(query={'search': f'name={pre_test_name}_org'})[0]
-    request.addfinalizer(org.delete)
-    sub = entities.Subscription(organization=org)
-    sub.refresh_manifest(data={'organization_id': org.id})
-    assert len(sub.search()) > 0
-    delete_manifest(org.name)
-    history = entities.Subscription(organization=org).manifest_history(
-        data={'organization_id': org.id}
-    )
-    assert "Subscriptions deleted by foreman_admin" == history[0]['statusMessage']
-
-
-@pre_upgrade
-def test_pre_subscription_scenario_autoattach(request):
-    """Create content host and register with Satellite
-
-    :id: preupgrade-940fc78c-ffa6-4d9a-9c4b-efa1b9480a22
-
-    :steps:
-        1. Before Satellite upgrade.
-        2. Create new Organization and Location
-        3. Upload a manifest in it.
-        4. Create a AK with 'auto-attach False' and without Subscription add in it.
-        5. Create a content host.
-        6. Update content host location.
-
-    :expectedresults:
-        1. Content host should be created.
-        2. Content host location should be updated.
+    The purpose this test scenario, to check  the auto-attached subscription on the client
```

Comment:
The scenario to test auto-attachment of subscription on the the client registered before upgrade.

Reply:
fixed.